src/AutoLayout/helpers.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react'
+import { getWidget } from './Column'
+import { useBreakpoints } from '../useBreakpoints'
+
+export function makeRows (children: any) {
+  const rows: any[] = []
+  let rowCount = 0
+
+  React.Children.forEach(children, (child, index) => {
+    if (!index && child) {
+      const newRow = [child]
+      rows.push(newRow)
+    }
+
+    if (child && child.props.newRow) {
+      rowCount += 1
+      const newRow = [child]
+      rows.push(newRow)
+    } else if (child && index) {
+      rows[rowCount].push(child)
+    }
+  })
+
+  return rows
+}
+
+export function wrapInWidget (rows: any, wrapInWidgets?: boolean) {
+  return rows.map((row: any) => {
+    return row.map((column: any, index: number) => {
+      const props = {
+        key: index,
+        grow: column.props.width ? undefined : 1,
+        height: '100%',
+        ...column.props
+      }
+
+      if (wrapInWidgets) return getWidget(props)
+
+      return React.cloneElement(column, props)
+    })
+  })
+}
+
+export function applySpacing (rows: any, heightOffSet: string, spacing: number | undefined) {
```

Comment:
`spacing?: number` would also be an option here

Reply:
Unfortunately not as im passing react props to it and beforehand you dont know whether its defined or not.